lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -62,116 +51,47 @@ namespace Intl
 
     bool IsWellFormedLanguageTag(_In_z_ const char16 *languageTag, _In_ const charcount_t cch)
     {
-        bool success = false;
         UErrorCode error = UErrorCode::U_ZERO_ERROR;
         char icuLocaleId[ULOC_FULLNAME_CAPACITY] = { 0 };
         char icuLangTag[ULOC_FULLNAME_CAPACITY] = { 0 };
         int32_t parsedLength = 0;
-        int32_t forLangTagResultLength = 0;
-        int32_t toLangTagResultLength = 0;
-        size_t inputLangTagUtf8SizeActual = 0;
-
-        // Allocate memory for the UTF8 output buffer. Need 3 bytes for each (code point + null) to satisfy SAL.
-        const size_t inputLangTagUtf8SizeAllocated = AllocSizeMath::Mul(AllocSizeMath::Add(cch, 1), 3);
-        unsigned char *inputLangTagUtf8 = new unsigned char[inputLangTagUtf8SizeAllocated];
-        if (!inputLangTagUtf8)
-        {
-            AssertOrFailFastMsg(false, "OOM: failed to allocate inputLangTagUtf8");
-            return false;
-        }
-
-        StringBufferAutoPtr<unsigned char> guard(inputLangTagUtf8);
-        memset(inputLangTagUtf8, 0, inputLangTagUtf8SizeAllocated); // explicitly zero the array
-
-        inputLangTagUtf8SizeActual = utf8::EncodeIntoAndNullTerminate(inputLangTagUtf8, languageTag, cch);
+        utf8::WideToNarrow languageTagUtf8(languageTag);
 
         // Convert input language tag to a locale ID for use in uloc_toLanguageTag API.
         // We used utf8 conversion to turn char16* into utf8char_t* (unsigned char *) but uloc_forLanguageTag takes char*
         // LangTags must be 7-bit-ASCII to be valid and any of these chars being "negative" is irrelevant.
-        forLangTagResultLength = uloc_forLanguageTag(reinterpret_cast<char *>(inputLangTagUtf8),
-            icuLocaleId, ULOC_FULLNAME_CAPACITY, &parsedLength, &error);
-        success = (forLangTagResultLength > 0) && (parsedLength > 0) &&
-            U_SUCCESS(error) && ((size_t)parsedLength == inputLangTagUtf8SizeActual);
-        if (!success)
-        {
-            return false;
-        }
+        int32_t forLangTagResultLength = uloc_forLanguageTag(languageTagUtf8, icuLocaleId, ULOC_FULLNAME_CAPACITY, &parsedLength, &error);
+        ICU_RETURN(error, forLangTagResultLength > 0 || parsedLength > 0 || ((size_t) parsedLength) == languageTagUtf8.Length(), false);
```

Comment:
why the change from && to || ?

Reply:
Because I had ICU_RETURN checking for true rather than false for a while and had to negate the entire boolean expression, but then I switched it back and only half-un-negated it. Good catch!